src/lib/portisWrapper.ts

```diff
@@ -0,0 +1,33 @@
+import { utf8ToHex } from 'web3-utils'
+
+const handler = {
+  get: function (target: any, prop: string) {
+    if (target[prop]) {
+      return target[prop]
+    }
+
+    if (prop === 'request') {
+      return function () {
+        const method = arguments[0].method
+
+        // chainId and accounts return as a Promise
+        if (method === 'eth_chainId' || method === 'eth_accounts' || method === 'net_version') {
+          return target.send(method)
+        }
+
+        // format params if using personal_sign
+        const params = (method === 'personal_sign')
+          ? [utf8ToHex(arguments[0].params[0]), arguments[0].params[1]]
+          : arguments[0].params
+
+        return new Promise((resolve, reject) => target.send(
+          { method, params },
+          (err: string | null, res: { result?: string, error?: Error }) =>
+            err ? reject(res.error) : resolve(res.result)
+        ))
+      }
+    }
+  }
+}
+
+export const portisWrapper = (provider: any) => new Proxy(provider, handler)
```

Comment:
Excellent! Some minor things:
- Should we add support for `eth_signTypedData` and `eth_sendTransaction` too? I don’t know if they work already
- Can we list the methods we tested?
- Let’s add some comments about this code

```suggestion
// This wrapper makes Portis Web SDK EIP-1193 compatible
export const portisWrapper = (provider: any) => new Proxy(provider, handler)
```

Reply:
I have added additional comments to the wrapper and added a section in the readme for Portis. Regarding `eth_signTypeData` it does not work out of the box as Portis' implementation is a bit different than Metamask's. Should I take the time now to implement this?